### ISSUE TITLE
@lodestar/prover package should support commonjs 

### ISSUE THREAD

**Issue by nazarhussain** [ChainSafe/lodestar#5755]
### Problem description

Currently the `@lodestar/prover` package have ESM build only, which does not work with the commonjs. Means user can't use `require` for that package in their projects. 

### Solution description

We should add support for the commonjs build for that package as it's a user facing library.

**Comment by nflaig** [comment]
I don't think we should put effort into supporting commonjs, there are [plenty of workarounds](https://stackoverflow.com/a/75287028/10577550) that can be used.

```ts
const prover = (await import('@lodestar/prover')).default
```

See [Pure ESM package](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) for more details.

**Comment by LucasGrasso** [comment]
I could find a workaround using `eval()` that works in jest also:
```typescript
/* eslint-disable no-eval */
const { ssz } = await eval('import("@lodestar/types")');
```
As mentioned in microsoft/TypeScript#43329

Most of lodestar packages are ESM packages so it worked with a bunch more of them. Hope it will do the trick for your issue.

**Comment by nazarhussain** [comment]
These workarounds work with top-level-await. And some environments does not support it.

**Comment by philknows** [comment]
What is the actual node.js roadmap for dual support? ESM seems to be the direction we are heading but has been slow. In the medium term this might be useful but requires investigation to create a low friction solution or dual module support.

**Comment by philknows** [comment]
Closing as we are no longer planning to support commonJS.
